connectors/standalone/src/test/scala/io/delta/standalone/internal/DeltaConfigSuite.scala

```diff
@@ -105,4 +112,52 @@ class DeltaConfigSuite extends FunSuite {
       assert(!isValidIntervalConfigValue(parseCalendarInterval(input)), s"$input")
     }
   }
+
+  test("Optional Calendar Interval config") {
+    val clock = new ManualClock(System.currentTimeMillis())
+    val emptyMetadata = new Metadata()
+    val metadataJ = MetadataJ.builder().schema(new StructType().add("part", new StringType())).build()
+    val metadata = ConversionUtils.convertMetadataJ(metadataJ)
+    val conf = new Configuration()
+
+    // case 1: duration not specified
+    withTempDir { dir =>
+      val log = DeltaLogImpl.forTable(conf, dir.getCanonicalPath, clock)
+      val retentionTimestampOpt = log.minSetTransactionRetentionInterval(emptyMetadata)
+
+      assert(retentionTimestampOpt.isEmpty)
+    }
+
+    // case 2: valid duration specified
+    withTempDir { dir =>
+      val log = DeltaLogImpl.forTable(conf, dir.getCanonicalPath, clock)
+      log.startTransaction().commit(
+        metadata.copy(
+          configuration = Map(DeltaConfigs.TRANSACTION_ID_RETENTION_DURATION.key -> "1 day")
+        ) :: Nil,
+        manualUpdate, writerId)
+
+
+      val retentionTimestampOpt = log.snapshot.minSetTransactionRetentionTimestamp
+      assert(log.clock.getTimeMillis() == clock.getTimeMillis())
+      val expectedRetentionTimestamp =
+        clock.getTimeMillis() - DeltaConfigs.getMilliSeconds(parseCalendarInterval("interval 1 days"))
+
+      assert(retentionTimestampOpt.contains(expectedRetentionTimestamp))
+    }
+
+    // case 3: invalid duration specified
+    withTempDir { dir =>
+      val e = intercept[IllegalArgumentException] {
+        val log = DeltaLogImpl.forTable(conf, dir.getCanonicalPath, clock)
+        log.startTransaction().commit(
+          metadata.copy(
+            configuration = Map(DeltaConfigs.TRANSACTION_ID_RETENTION_DURATION.key -> "interval 1 foo")
+          ) :: Nil,
+          manualUpdate, writerId)
+        log.snapshot.minSetTransactionRetentionTimestamp // TODO: This step required to get the exception
```

Comment:
Is there any other way to trigger "validateConfigurations" as the intention of this line is trigger the validation to get the exception.